### ISSUE TITLE
Comparing basemap and cartopy

### ISSUE THREAD

**Issue by guziy** [SciTools/cartopy#525]
Hi:

I was checking out cartopy since basemap author told that this is future....

Can someone tell me if I can do a similar plot with cartopy? 
![image](https://cloud.githubusercontent.com/assets/900941/5570609/6d9016c4-8f52-11e4-9c61-572bd307f1fb.png)

Below is my effort to do it with cartopy, which does not seem to work very well.
![image](https://cloud.githubusercontent.com/assets/900941/5570601/4143b774-8f52-11e4-8f3e-fb13ff8980e3.png)

The code for the images can be found here:
http://nbviewer.ipython.org/github/guziy/PyNotebooks/blob/master/cartopy_compare_with_basemap.ipynb

Thanks

**Comment by pelson** [comment]
You might be interested in the docstring (not yet in the latest release) at https://github.com/SciTools/cartopy/blob/master/lib/cartopy/crs.py#L1036.

I think the CRS you want is:

```
rpole_crs = ccrs.RotatedPole(
                             pole_longitude=bmap_params["lon_0"] + 180, 
                             pole_latitude=bmap_params["o_lat_p"],
                             )
```

Notice that the o_lon_p is not being used here. The reason is that the RotatedPole in cartopy has been implemented to work well with meteorological type projections where a pole rotation is used to provide an approximately equidistant limited area domain. The meteorological definition of rotated pole (as documented by the WMO) specifies the latitude longitude rotation in a different order to the way proj4 has implemented it, therefore cartopy moved to ratify the problem, hence the confusion here (esp. if you are used to the proj4 definition). For more information see the thread entitled "proj4 specifications for a rotated latlon grid on a sphere?" [here](http://lists.maptools.org/pipermail/proj/2012-August/006391.html).

HTH

**Comment by guziy** [comment]
Thanks Pelson. 

Still it looks different...

Do you know what is +180 for in the projection definition?

I also get wrong result when trying to set extent using projection coordinates.... Can you tell me what I am doing wrong? (In [13] - here: http://nbviewer.ipython.org/github/guziy/PyNotebooks/blob/master/cartopy_compare_with_basemap.ipynb)

Is there a set_corners function?

Here are some of my experiments

http://nbviewer.ipython.org/github/guziy/PyNotebooks/blob/master/cartopy_compare_with_basemap.ipynb

By the way I was not able to use pip version in ipython, so I had to install the github version of cartopy. 
Here is the warning I was getting instead of the plot (in the same nb where baseman and simple mpl worked just fine):
<pre>
/Users/huziy/PyVirtualenvs/basic-py2.7/lib/python2.7/site-packages/IPython/core/formatters.py:239: FormatterWarning: Exception in image/png formatter: HTTP Error 404: Not Found
  FormatterWarning,
</pre>

Cheers

**Comment by pelson** [comment]
> I also get wrong result when trying to set extent using projection coordinates.... Can you tell me what I am doing wrong?

The set_extent method is making a box with the coordinates you are asking for, and then projecting that box back into the target projection. Therefore set_extent is fundamentally not the same as setting the corners as you get with Basemap. To see what is going on here, it is useful to take a look at the box itself:

```
import cartopy.crs as ccrs
import matplotlib.pyplot as plt
import shapely.geometry as sgeom

box = sgeom.box(minx=-90, maxx=45, miny=15, maxy=70)

proj = ccrs.RotatedPole(
                             pole_longitude=bmap_params["lon_0"] + 180, 
                             pole_latitude=bmap_params["o_lat_p"],
                             )

ax1 = plt.axes(projection=proj)
ax1.coastlines()
ax1.add_geometries([box], ccrs.PlateCarree(), facecolor='coral', 
                                  edgecolor='black', alpha=0.5)

```


> By the way I was not able to use pip version in ipython

That is a weird one on me - I suspect there is a traceback somewhere which we can get hold of. It may be that the traceback can tell us more about the installation problem.

**Comment by pelson** [comment]
Cool. Thanks for that @ajdawson - this looks pretty comparable now. :+1:

**Comment by guziy** [comment]
It looks great)) Thanks a lot for your replies

Cheers

2015-01-09 4:10 GMT-05:00 Andrew Dawson <notifications@github.com>:

> @guziy <https://github.com/guziy> - there are two very small errors in
> your notebook when setting the extent in projection coordinates:
>
>    1. Your projection should be defined with pole_longitude=bmap_params["lon_0"]
>    + 180, whereas you have pole_longitude=bmap_params["o_lon_p"] + 180.
>    2. You have the latitude and longitude the wrong way round in the
>    projection of the corners, it should be
>
>     xll, yll = rpole_crs.transform_point(bmap_params["llcrnrlon"], bmap_params["llcrnrlat"], ccrs.Geodetic())
>     xur, yur = rpole_crs.transform_point(bmap_params["urcrnrlon"], bmap_params["urcrnrlat"], ccrs.Geodetic())
>
> When I changed these two things I got the following result, which looks
> correct I think:
>
> [image: rpole]
> <https://cloud.githubusercontent.com/assets/390602/5677478/2627f55a-97df-11e4-888d-bb4bc4a29267.png>
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/SciTools/cartopy/issues/525#issuecomment-69309732>.
>



-- 
Sasha

**Comment by guziy** [comment]
To @pelson:
There was no traceback in ipython. But here is what I get in command line: 
<pre>
/Users/huziy/virtualenvs/py2.7-default/lib/python2.7/site-packages/cartopy/io/__init__.py:261: DownloadWarning: Downloading: http://www.nacis.org/naturalearth/110m/physical/ne_110m_coastline.zip
  warnings.warn('Downloading: {}'.format(url), DownloadWarning)
Traceback (most recent call last):
  File "/Users/huziy/virtualenvs/py2.7-default/lib/python2.7/site-packages/matplotlib/artist.py", line 59, in draw_wrapper
    draw(artist, renderer, *args, **kwargs)
  File "/Users/huziy/virtualenvs/py2.7-default/lib/python2.7/site-packages/matplotlib/figure.py", line 1079, in draw
    func(*args)
  File "/Users/huziy/virtualenvs/py2.7-default/lib/python2.7/site-packages/matplotlib/artist.py", line 59, in draw_wrapper
    draw(artist, renderer, *args, **kwargs)
  File "/Users/huziy/virtualenvs/py2.7-default/lib/python2.7/site-packages/cartopy/mpl/geoaxes.py", line 355, in draw
    inframe=inframe)
  File "/Users/huziy/virtualenvs/py2.7-default/lib/python2.7/site-packages/matplotlib/artist.py", line 59, in draw_wrapper
    draw(artist, renderer, *args, **kwargs)
  File "/Users/huziy/virtualenvs/py2.7-default/lib/python2.7/site-packages/matplotlib/axes/_base.py", line 2092, in draw
    a.draw(renderer)
  File "/Users/huziy/virtualenvs/py2.7-default/lib/python2.7/site-packages/matplotlib/artist.py", line 59, in draw_wrapper
    draw(artist, renderer, *args, **kwargs)
  File "/Users/huziy/virtualenvs/py2.7-default/lib/python2.7/site-packages/cartopy/mpl/feature_artist.py", line 103, in draw
    geoms = self._feature.intersecting_geometries(extent)
  File "/Users/huziy/virtualenvs/py2.7-default/lib/python2.7/site-packages/cartopy/feature.py", line 113, in intersecting_geometries
    return (geom for geom in self.geometries() if
  File "/Users/huziy/virtualenvs/py2.7-default/lib/python2.7/site-packages/cartopy/feature.py", line 180, in geometries
    name=self.name)
  File "/Users/huziy/virtualenvs/py2.7-default/lib/python2.7/site-packages/cartopy/io/shapereader.py", line 246, in natural_earth
    return ne_downloader.path(format_dict)
  File "/Users/huziy/virtualenvs/py2.7-default/lib/python2.7/site-packages/cartopy/io/__init__.py", line 223, in path
    result_path = self.acquire_resource(target_path, format_dict)
  File "/Users/huziy/virtualenvs/py2.7-default/lib/python2.7/site-packages/cartopy/io/shapereader.py", line 301, in acquire_resource
    shapefile_online = self._urlopen(url)
  File "/Users/huziy/virtualenvs/py2.7-default/lib/python2.7/site-packages/cartopy/io/__init__.py", line 262, in _urlopen
    return urlopen(url)
  File "/opt/local/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/urllib2.py", line 127, in urlopen
    return _opener.open(url, data, timeout)
  File "/opt/local/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/urllib2.py", line 410, in open
    response = meth(req, response)
  File "/opt/local/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/urllib2.py", line 523, in http_response
    'http', request, response, code, msg, hdrs)
  File "/opt/local/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/urllib2.py", line 442, in error
    result = self._call_chain(*args)
  File "/opt/local/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/urllib2.py", line 382, in _call_chain
    result = func(*args)
  File "/opt/local/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/urllib2.py", line 629, in http_error_302
    return self.parent.open(new, timeout=req.timeout)
  File "/opt/local/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/urllib2.py", line 410, in open
    response = meth(req, response)
  File "/opt/local/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/urllib2.py", line 523, in http_response
    'http', request, response, code, msg, hdrs)
  File "/opt/local/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/urllib2.py", line 448, in error
    return self._call_chain(*args)
  File "/opt/local/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/urllib2.py", line 382, in _call_chain
    result = func(*args)
  File "/opt/local/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/urllib2.py", line 531, in http_error_default
    raise HTTPError(req.get_full_url(), code, msg, hdrs, fp)
HTTPError: HTTP Error 404: Not Found
</pre>

This has happened on 2 machines with yosemite.

Cheers

**Comment by pelson** [comment]
Yep - Natural Earth changed where they host their data again. v0.11.x and master both have this fixed as a result of #532.
